database/src/lib.rs

```diff
@@ -798,3 +798,154 @@ pub struct ArtifactCollection {
     pub duration: Duration,
     pub end_time: DateTime<Utc>,
 }
+
+#[derive(Debug)]
+pub enum BenchmarkRequestCommitType {
+    Try { pr: u32 },
+    Master { pr: u32 },
+    Release { tag: String },
+}
+
+impl fmt::Display for BenchmarkRequestCommitType {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            BenchmarkRequestCommitType::Try { pr: _ } => write!(f, "try"),
+            BenchmarkRequestCommitType::Master { pr: _ } => write!(f, "master"),
+            BenchmarkRequestCommitType::Release { tag: _ } => write!(f, "release"),
+        }
+    }
+}
+
+#[derive(Debug)]
+pub enum BenchmarkRequestStatus {
+    WaitingForArtifacts,
+    WaitingForParent,
+    Queued,
+    InProgress,
+    Completed,
+}
+
+impl fmt::Display for BenchmarkRequestStatus {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            BenchmarkRequestStatus::WaitingForArtifacts => write!(f, "waiting_for_artifacts"),
+            BenchmarkRequestStatus::WaitingForParent => write!(f, "waiting_for_parent"),
+            BenchmarkRequestStatus::Queued => write!(f, "queued"),
+            BenchmarkRequestStatus::InProgress => write!(f, "in_progress"),
+            BenchmarkRequestStatus::Completed => write!(f, "completed"),
+        }
+    }
+}
+
+#[derive(Debug)]
+pub enum BenchmarkRequestType {
+    /// A Try commit
+    Try {
+        sha: String,
+        parent_sha: Option<String>,
+        pr: u32,
+    },
+    /// A Master commit
+    Master {
+        sha: String,
+        parent_sha: Option<String>,
+        pr: u32,
+    },
+    /// A release only has a tag
+    Release { tag: String },
+}
+
+impl fmt::Display for BenchmarkRequestType {
```

Comment:
I know that we sometimes `impl Display` so that we can just YOLO certain structs into a single TEXT column of a DB, which is already kind of an abuse of the `Display` trait, but here the display trait purposefully ignores certain fields, which seems like too much of a hack. Let's just add a method on the struct, such as `commit_type(&self) -> &str` that will return the commit type string.